### ISSUE TITLE
i18n

### ISSUE THREAD

**Issue by notramo** [plentico/plenti#267]
Design a user-friendly i18n solution which is integrated with the Plenti CMS, but also supported by Weblate.

**Comment by jimafisk** [comment]
Thanks for the suggestion @notramo! Can you explain a little more about how you envision this working? I don't know too much about Weblate, but it appears to be a python program that would need to be hosted. Is there an SDK you'd like to integrate with the editing interface, or does Weblate produce JSON that the Plenti SSG could use as a content source?

You can support different languages currently in your content source, but it wouldn't have any of the advanced features I'm sure Weblate provides. I made a [small example site](https://github.com/jimafisk/plenti-multilingual-example) a while back, but there are many different ways you could accomplish the same thing.

In the future when you can make your own custom editor widgets (https://github.com/plentico/plenti/issues/219), you could build all sorts of features to support i18n.

**Comment by notramo** [comment]
It wouldn't be specific to Weblate, all translation platforms have similar requirements, but Weblate is the most advanced open source one. Also, they offer a hosted version which is free for open source projects.

When an editor adds a new string, they shouldn't have to care about how many languages it is translated to. They just add the string in their main language, and the translation platform should do the rest.
When they change a string, they shouldn't have to bother changing it in every translation, as Weblate tracks it automatically.

Thanslation files can be these two formats (I don't know the exach schema, but it's enough for demonstration.):
```json
{
  "key1": "translation string",
  "key2": "another string"
}
```
```json
{
  "English string": "localized string",
  "another English string": "another localized string"
}
```

So basically Weblate (and other translation platforms) require a specific JSON schema.

The UX in the CMS should be the ideally same as non-localized sites, or probably add a checkbox next to the string inputs with “translatable” label (to prevent translating some strings).

**Comment by jimafisk** [comment]
Hi @blahetal, sorry this slipped through the cracks! Yes editing the page in each language individually should work like you said.

I just added a quick login button and [pushed the changes to the example site](https://github.com/jimafisk/plenti-multilingual-example/commit/5b7206be5ba83d6711a5f82a47a5b03e144a6ff1) if you want to take a look:

<details>
<summary>Click to see gif</summary>

![multilingual2](https://github.com/plentico/plenti/assets/5913244/b6baed44-4f6d-4d4c-9b45-dd483cfdad6a)


</details>
